src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs

```diff
@@ -1263,27 +1264,21 @@ private int OleGetDataUnrestricted(ref FORMATETC formatetc, ref STGMEDIUM medium
         /// <summary>
         /// Retrieves a list of distinct strings from the array.
         /// </summary>
-        private static string[] GetDistinctStrings(string[] formats)
+        private static string[] GetDistinctStrings(List<string> formats)
```

Comment:
How about `formats.Distinct().ToArray()` ? Which is faster?

Reply:
@lindexi Distinct will use internally HashSet<T>, which as I've mentioned, don't think is suitable for this usage. It usually comes out on top with 14 or more objects but even then will allocate double the memory just to build it.